src/tests/class_tests/openms/source/NASequence_test.cpp

```diff
@@ -0,0 +1,751 @@
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Samuel Wein $
+// $Authors: Timo Sachsenberg, Samuel Wein $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/CONCEPT/ClassTest.h>
+#include <OpenMS/test_config.h>
+
+///////////////////////////
+#include <OpenMS/CHEMISTRY/NASequence.h>
+#include <iostream>
+#include <OpenMS/SYSTEM/StopWatch.h>
+#include <OpenMS/CHEMISTRY/Ribonucleotide.h>
+#include <OpenMS/CHEMISTRY/RibonucleotideDB.h>
+
+///////////////////////////
+
+using namespace OpenMS;
+using namespace std;
+
+START_TEST(NASequence, "$Id$")
+
+/////////////////////////////////////////////////////////////
+/////////////////////////////////////////////////////////////
+
+NASequence* ptr = nullptr;
+NASequence* null_ptr = nullptr;
+RibonucleotideDB* db = RibonucleotideDB::getInstance();
+
+START_SECTION((NASequence()=default))
+{
+  ptr = new NASequence();
+  TEST_NOT_EQUAL(ptr, null_ptr)
+  TEST_EQUAL(ptr->getFivePrimeMod(), (void*) nullptr);
+  TEST_EQUAL(ptr->getThreePrimeMod(), (void*) nullptr);
+  TEST_EQUAL(ptr->size(), 0);
+}
+END_SECTION
+
+
+START_SECTION((NASequence(const NASequence&) = default))
+{
+  // test Copy Constructor
+  NASequence aaa = NASequence::fromString("AAA");
+  NASequence aaa2(aaa);
+
+  TEST_EQUAL(aaa.size(), 3);
+  TEST_EQUAL(aaa2.size(), 3);
+  TEST_EQUAL(aaa.operator==(aaa2), true);
+}
+END_SECTION
+
+START_SECTION((NASequence(NASequence&&) = default))
+{
+  // test Move constructor
+  NASequence aaa(NASequence::fromString("AAA"));
+  TEST_EQUAL(aaa.size(), 3);
+}
+END_SECTION
+
+START_SECTION((NASequence& operator=(const NASequence&)& = default))
+{
+  // test Copy Assignment
+  NASequence aaa = NASequence::fromString("AAA");
+  NASequence c = NASequence::fromString("C");
+  c = aaa;
+  TEST_EQUAL(aaa.size(), 3);
+  TEST_EQUAL(c.size(), 3);
+  TEST_EQUAL(aaa.operator==(c), true);
+}
+END_SECTION
+
+START_SECTION((NASequence& operator=(NASequence&&)& = default))
+{
+  // test Move Assignment
+  NASequence c = NASequence::fromString("C");
+  c = NASequence::fromString("AAA");
+  TEST_EQUAL(c.size(), 3);
+}
+END_SECTION
+
+START_SECTION((NASequence(std::vector<const Ribonucleotide*> s, const RibonucleotideChainEnd* five_prime, const RibonucleotideChainEnd* three_prime)))
+{
+  NASequence aaa = NASequence::fromString("AAA");
+  NASequence aaa2(aaa.getSequence(), nullptr, nullptr);
```

Comment:
can you simplify this by having another constructor `explicit NASequence(std::vector<const Ribonucleotide*> s,` ?